velox/common/memory/ByteStream.cpp

```diff
@@ -17,9 +17,44 @@
 #include "velox/common/memory/ByteStream.h"
 
 namespace facebook::velox {
+
+std::streampos ByteStream::tellp() const {
+  if (ranges_.empty()) {
+    return 0;
+  }
+  assert(current_);
+  int64_t size = 0;
+  for (auto& range : ranges_) {
+    if (&range == current_) {
+      return current_->position + size;
+    }
+    size += range.numValues();
+  }
+  VELOX_FAIL("ByteStream 'current_' is not in 'ranges_'.");
+}
+
+void ByteStream::seekp(std::streampos position) {
+  int64_t toSkip = position;
+  if (current_ && current_->position > current_->fill) {
+    current_->fill = current_->position;
```

Comment:
I find this logic confusing. Shouldn't "fill" be updated when "position" is updated?